packages/kbn-i18n/src/core/i18n.test.js

```diff
@@ -0,0 +1,737 @@
+    });
+
+    test('should return updated locale', () => {
+      i18n.setDefaultLocale('foo');
+      expect(i18n.getDefaultLocale()).toBe('foo');
+    });
+  });
+
+  describe('setFormats', () => {
+    test('should throw error if formats parameter is not an object', () => {
+      expect(() => i18n.setFormats(undefined)).toThrow();
+      expect(() => i18n.setFormats(null)).toThrow();
+      expect(() => i18n.setFormats(true)).toThrow();
+      expect(() => i18n.setFormats(5)).toThrow();
+      expect(() => i18n.setFormats('foo')).toThrow();
+    });
+
+    test('should throw error if formats parameter is empty object', () => {
+      expect(() => i18n.setFormats({})).toThrow();
+    });
+
+    test('should merge current formats with a passed formats', () => {
+      expect(i18n.getFormats().date.short).not.toEqual({
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      });
+
+      i18n.setFormats({
+        date: {
+          short: {
+            month: 'short',
+            day: 'numeric',
+            year: 'numeric',
+          },
+        },
+      });
+
+      expect(i18n.getFormats().date.short).toEqual({
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      });
+
+      i18n.setFormats({
+        date: {
+          short: {
+            month: 'long',
+            day: 'numeric',
+            year: 'numeric',
+          },
+        },
+      });
+
+      expect(i18n.getFormats().date.short).toEqual({
+        month: 'long',
+        day: 'numeric',
+        year: 'numeric',
+      });
+    });
+  });
+
+  describe('getFormats', () => {
+    test('should return "en" formats by default', () => {
+      const { formats } = require('./formats');
+
+      expect(i18n.getFormats()).toEqual(formats);
+    });
+
+    test('should return updated formats', () => {
+      const { formats } = require('./formats');
+
+      i18n.setFormats({
+        foo: 'bar',
+      });
+
+      expect(i18n.getFormats()).toEqual({
+        ...formats,
+        foo: 'bar',
+      });
+    });
+  });
+
+  describe('getRegisteredLocales', () => {
+    test('should return empty array by default', () => {
+      expect(i18n.getRegisteredLocales()).toEqual([]);
+    });
+
+    test('should return array of registered locales', () => {
+      i18n.addMessages({
+        locale: 'en',
+      });
+
+      expect(i18n.getRegisteredLocales()).toEqual(['en']);
+
+      i18n.addMessages({
+        locale: 'ru',
+      });
+
+      expect(i18n.getRegisteredLocales()).toContain('en', 'ru');
+      expect(i18n.getRegisteredLocales().length).toBe(2);
+
+      i18n.addMessages({
+        locale: 'fr',
+      });
+
+      expect(i18n.getRegisteredLocales()).toContain('en', 'ru', 'fr');
+      expect(i18n.getRegisteredLocales().length).toBe(3);
+    });
+  });
+
+  describe('translate', () => {
+    test('should throw error if id is not a string', () => {
+      expect(() => i18n.translate(undefined)).toThrow();
```

Comment:
__issue:__ since `translate` is the most important function in this framework and developers will mostly see errors originating from this function let's use `toThrowErrorMatchingSnapshot` in __all__ tests in this suite to make sure we generate what we expect.